u_public.go

```diff
@@ -38,17 +41,20 @@ type PubClientHandshakeState struct {
 
 // TLS 1.3 only
 type TLS13OnlyState struct {
-	Suite           *PubCipherSuiteTLS13
+	// Deprecated: Use KeyShareKeys instead.
 	EcdheKey        *ecdh.PrivateKey
 	KeySharesParams *KeySharesParameters
```

Comment:
Should we be removing these old versions? If not (for backward compatibility), we should document here what happens when both KeySharePrivateKeys and these deprecated ones are provided.

Reply:
Good point! I added some docs about when both are set.